### DISCUSSION TITLE
libp2p-rPGF-1: Nominations for Education, Outreach, and Applications

### DISCUSSION THREAD

**Discussion by p-shahi** [libp2p/libp2p#212] [RetroPGF]
### Education, Outreach, and Applications

This discussion post is for projects nominations to the  Education, Outreach, and Applications category.
This category is intended for projects and works improving libp2p educational material, documentation, writing sample applications, etc.

### Nomination Template
When nominating a project, please answer the following questions in a comment to this discussion:

```
Project name:
The project name

Description:
A description of the positive impact of this project for this specific category over the past year (March 2023 - March 2024).

Github/Twitter link:
A link to the project’s GitHub or Twitter

Contact:
Contact info for the project or project lead
```

Things to consider when nominating a project:

- Describe how this project has delivered impact to the libp2p ecosystem in the previous year (March 2023 - March 2024). Use the provided template.
- Nominate projects in **one** relevant category. Do not nominate the same project across different categories! If a project has overlap in multiple categories, please choose just one category that best aligns with it. 
- Organizers of libp2p-RPGF-1 may reassign a project’s category from its nominated one if it better aligns elsewhere.
- Check if a project has been nominated already to avoid duplication.
- Please provide feedback to help the process improve in each round. Your input and feedback is valuable and you can use this post to submit feedback now.

Nominations will close on May 3rd, 2024.

See https://github.com/libp2p/libp2p/discussions/209 for an overview of the process, timelines, and to ask general questions.

**Comment by TimDaub** [comment]
## Project name

Kiwi News

## Description

### Introduction

Kiwi News (https://kiwinews.xyz) is a crypto HackerNews app built on top of a set reconciliation algorithm implemented with libp2p.

Kiwi News lets users submit, upvote, and comment on links, just like Hacker News. Most of the links are focused on crypto tech, products, and culture. 

Kiwi News Protocol implements a similar set reconcilation algorithm as Farcaster. Node operators can replicate the network’s entire database and spin up their own clients or fork if they are unhappy with, for example, our style of moderation.

### Impact

- We successfully replicated the Farcaster whitepaper in April 2023 with a working implementation of the protocol and demonstrated that their architecture claims about decentralization are, indeed, valid.
- We have given several educational talks at notable crypto conferences about this set reconciliation algorithm, among them Libp2p Day in Istanbul, Zusocial in Istanbul, Protocol Berg in Berlin, and Zeitgeist’s Demo Day in New York. Find a list of all the talks’ recordings below:
  - https://www.youtube.com/watch?v=Rys5UEi2SWg
  - https://www.youtube.com/watch?v=WujtU15yAyk
  - https://www.youtube.com/watch?v=1U7tXX2HxOk
  - https://www.youtube.com/watch?v=d63iPJ_2u5Q
- We wrote the first in-depth documentation that details the entirety of Farcaster’s and Kiwi News’s set reconciliation algorithm, making our months-long research accessible to anyone: https://attestate.com/kiwistand/main/protocol-guide.html

**Apart from this, we would like you to consider the following:**

Strategically, we've tried to popularize the set reconciliation algorithm by building an amazing consumer product. In that way, we align with Farcaster and Warpcast’s strategy.

We did this because we’re convinced that only by increasing our Daily Active Users will we be able to truly show the power and usability of libraries like libp2p, libp2p-gossipsub, etc.

Therefore, please also consider taking a look at our product, which is accessible to consumers but built on a solid, decentralized p2p foundation without burdening the consumer too much with this information.

Product: Kiwinews.xyz

## Github/Twitter link:

- GitHub: https://github.com/attestate/kiwistand
- Twitter: https://twitter.com/KiwiNewsHQ 

## Contact: 
@timdaub on Telegram | tim@daubenschuetz.de

**Comment by 2color** [comment]
## Project name: 

**Universal Connectivity**

## Description:

Realtime decentralised chat with libp2p showing ubiquitous peer-to-peer connectivity between multiple programming languages (Go, Rust, TypeScript) and runtimes (Web, binary) using QUIC, WebRTC and WebTransport.

## Talks & Impact

The project has helped teach developers many of the basic patterns of libp2p across different language implementations. It has also gotten many community PRs adding file transfers. Currently it has 66 stars and 27 forks on GitHub. It also served as a catalyst for https://github.com/libp2p/specs/discussions/587 which became a [spec proposal](https://github.com/libp2p/specs/pull/590) which I began [experimenting with its implementation](https://github.com/libp2p/js-libp2p/pull/2254). In a sense it serves as a incubator to the end-to-end developer experience with libp2p. 

https://www.youtube.com/watch?v=9qE6JE_GN_U
https://www.youtube.com/watch?v=4v-iIB0C9_8

A description of the positive impact of this project for this specific category over the past year (March 2023 - March 2024).

Github/Twitter link:

- https://github.com/libp2p/universal-connectivity

Contact:
- @2color `daniel [at] ipshipyard.com`

**Comment by cewood** [comment]
**Project name:**
Amino DHT Bootstrappers

**Description:**
In order for a libp2p node to join the Amino DHT, it is necessary for it to discover other peers, and bootstrap nodes facilitate this process. The IPFS Foundation, with the help of the Interplanetary Shipyard, provides several public bootstrap nodes that are published as a TXT record to DNS at _dnsaddr.bootstrap.libp2p.io 

**Github/Twitter link:**
https://docs.ipfs.tech/concepts/public-utilities/#amino-dht-bootstrappers 

**Contact:**
[@aschmahmann](https://github.com/aschmahmann) 
[@cewood](https://github.com/cewood) 
[@lidel](https://github.com/lidel)
